// @vercel/hono/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  build: () => build,
  findEntrypoint: () => findEntrypoint,
  shim: () => shim,
  shouldServe: () => shouldServe,
  startDevServer: () => startDevServer,
  version: () => version
});
module.exports = __toCommonJS(src_exports);

// src/build.ts
var import_node = require("@vercel/node");
var build = async (args) => {
  const entrypoint = findEntrypoint(args.files);
  return (0, import_node.build)({
    ...args,
    entrypoint,
    useWebApi: true,
    shim
  });
};
var shim = (handler, relativePathToHandler = ".") => `
// @ts-ignore
import app from "${relativePathToHandler}/${handler}";

// @ts-ignore
const handle = async (request) => {
  return app.fetch(request);
};

export const GET = handle;
export const POST = handle;
export const PUT = handle;
export const DELETE = handle;
export const PATCH = handle;
export const OPTIONS = handle;
export const HEAD = handle;`;
var findEntrypoint = (files) => {
  const validEntrypoints = [
    "server.ts",
    "server.js",
    "index.ts",
    "index.js",
    "index.mjs",
    "index.cjs",
    "src/index.ts",
    "src/index.js",
    "src/index.mjs",
    "src/index.cjs"
  ];
  const entrypoint = validEntrypoints.find((path) => files[path] !== void 0);
  if (!entrypoint) {
    throw new Error("No valid entrypoint found");
  }
  return entrypoint;
};

// src/index.ts
var import_node2 = require("@vercel/node");
var import_promises = require("fs/promises");
var import_path = require("path");
var version = 3;
var shouldServe = async (opts) => {
  const requestPath = opts.requestPath.replace(/\/$/, "");
  if (requestPath.startsWith("api")) {
    return false;
  }
  return true;
};
var startDevServer = async (opts) => {
  const entrypoint = findEntrypoint(opts.files);
  const entrypointExtension = entrypoint.split(".").pop();
  const shimString = shim(entrypoint, "../..");
  const shimEntrypoint = `.vercel/dev/shim.${entrypointExtension}`;
  const shimEntrypointPath = (0, import_path.join)(opts.workPath, shimEntrypoint);
  const shimEntrypointDir = (0, import_path.dirname)(shimEntrypointPath);
  await (0, import_promises.rm)(shimEntrypointDir, { force: true, recursive: true });
  await (0, import_promises.mkdir)(shimEntrypointDir, { recursive: true });
  await (0, import_promises.writeFile)(shimEntrypointPath, shimString);
  return (0, import_node2.startDevServer)({
    ...opts,
    entrypoint: shimEntrypoint
  });
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  build,
  findEntrypoint,
  shim,
  shouldServe,
  startDevServer,
  version
});
